test/src/kbasesearchengine/test/main/NarrativeInfoDecoratorTest.java

```diff
@@ -372,6 +377,66 @@ public void listTypes() throws Exception {
                 "type", dummyOutput)));
     }
 
+    @Test
+    public void searchObjectsWithDeletedWs() throws Exception {
+        final NarrativeInfoDecorator nid = setUpSearchObjectsWithDeletedWs();
+        final Pagination pag = new Pagination().withCount(3L).withStart(0L);
+        final SearchObjectsInput dummyInput = new SearchObjectsInput()
+                .withPostProcessing(new PostProcessing().withAddNarrativeInfo(1L))
+                .withPagination(pag);
+
+        final ArrayList<ObjectData> expectedObjs = new ArrayList<>();
+        expectedObjs.add(new ObjectData().withGuid("WS:4/1/7").withCreator("user"));
+        expectedObjs.add(new ObjectData().withGuid("WS:4/2/1").withCreator("user"));
+        expectedObjs.add(new ObjectData().withGuid("WS:4/1/7").withCreator("user"));
```

Comment:
Shouldn't there be a WS:3/5/6 because setUpSearchObjectsWithDeletedWs() has this as one of the three guids?

I don't follow how this test is passing.

Reply:
anything from WS:/3 should be removed as it is shown to be deleted. Currently they are returned as null, which is a problem as the front end sometimes gets pages of deleted workspaces.